### ISSUE TITLE
Python 3.7 compatibility issue introduced by release 1.3.0

### ISSUE THREAD

**Issue by nicolas-seichepine** [joblib/joblib#1469]
Hi,

If I understand correctly, latest release [should work](https://github.com/joblib/joblib/blob/b9be27d7ce405417572fee6fcd9317c6b6ff67b6/pyproject.toml#L19) with python 3.7

In the meantime, [this specific line](https://github.com/joblib/joblib/blob/b9be27d7ce405417572fee6fcd9317c6b6ff67b6/joblib/externals/loky/backend/context.py#L21) introduced in release 1.3.0 breaks compatibility (`ImportError: cannot import name '_MAX_WINDOWS_WORKERS' from 'concurrent.futures.process'`)

**Comment by imatiach-msft** [comment]
I'm seeing a similar error on windows python 3.7:

  File "C:\Miniconda\envs\test\lib\site-packages\joblib\externals\loky\backend\popen_loky_win32.py", line 153, in main
    _winapi.SYNCHRONIZE | _winapi.PROCESS_DUP_HANDLE, False, parent_pid
AttributeError: module '_winapi' has no attribute 'SYNCHRONIZE'

trying to pin joblib < 1.3.0 in the tests for now...

**Comment by glutamatt** [comment]
some hints :
![image](https://github.com/joblib/joblib/assets/1442992/f1769c01-85bf-4c23-886a-60f475a0cf3e)
![image](https://github.com/joblib/joblib/assets/1442992/060d702e-d00c-4fcf-9feb-c148043b8ea2)

**Comment by tomMoral** [comment]
Thanks a lot for the report and investigations.
I opened a PR to fix this in `loky` (joblib/loky#408). I will try to do a bug fix release asap, feel free to let us know if anything else come up before this.

**Comment by imatiach-msft** [comment]
Seems my problem is a bit different, I created a new issue here:

https://github.com/joblib/loky/issues/411

**Comment by tomMoral** [comment]
The problem should be resolved on unix platform with the release 1.3.1.
Let us know if more problem persists.

For windows user, a potential way around is to use the `spawn` processes in `loky, which should provide a relatively good support but might still have some issues with controlling the C-level parallelism for instance with numpy.

```python
from joblib.externals.loky.backend.context import set_start_method

set_start_method('spawn')
```

**Comment by imatiach-msft** [comment]
I just pinned to joblib<1.3.0 on python 3.7 & windows platform and it works for now.  I think most of our users are on unix anyway but we like to exhaustively test every python/platform/oss package combination.
